lib/brains/forms.py

```diff
@@ -110,3 +128,11 @@ def braintree_data(self):
             self.cleaned_data['bt_signature'],
             self.cleaned_data['bt_payload']
         )
+
+    def clean(self):
+        res = requests.post(settings.BRAINTREE_PROXY + '/parse', self.data)
+        if res.status_code != 204:
+            log.error('Did not receive a 204 from solitude-auth, got: {}'
+                      .format(res.status_code))
+            raise forms.ValidationError('Did not pass verification')
```

Comment:
Since this is the point at which we can trust the payload, it might be more convenient to unpack it here:

    data = base64.decodestring(self.cleaned_data['bt_payload'])
    data = XmlUtil.dict_from_xml(data)
    self.cleaned_data['bt_payload'] = data